src/app/admin/createCollectionCentre/CreateCollectionCentreForm.tsx

```diff
@@ -0,0 +1,103 @@
+import RefreshPageButton from "@/app/admin/RefreshPageButton";
+import { Schema } from "@/database_utils";
+import NameCard from "@/app/admin/createCollectionCentre/createCollectionCentreFormSections/NameCard";
+import AcronymCard from "@/app/admin/createCollectionCentre/createCollectionCentreFormSections/AcronymCard";
+import supabase from "@/supabaseClient";
+
+const initialFields: Schema["collection_centres"] = {
+    name: "",
+    acronym: "",
+};
+
+const initialFormErrors: FormErrors = {
+    name: Errors.initial,
+    acronym: Errors.initial,
+};
+
+const formSections = [NameCard, AcronymCard];
+
+const CreateCollectionCentreForm: React.FC<{}> = () => {
+    const [fields, setFields] = useState(initialFields);
+    const [formErrors, setFormErrors] = useState(initialFormErrors);
+
+    const fieldSetter = setField(setFields, fields);
+    const errorSetter = setError(setFormErrors, formErrors);
+
+    const [submitError, setSubmitError] = useState(Errors.none);
+    const [submitErrorMessage, setSubmitErrorMessage] = useState("");
+    const [submitDisabled, setSubmitDisabled] = useState(false);
+
+    const [refreshRequired, setRefreshRequired] = useState(false);
+
+    const submitForm = async (): Promise<void> => {
+        setSubmitDisabled(true);
+
+        if (checkErrorOnSubmit(formErrors, setFormErrors)) {
+            setSubmitError(Errors.submit);
+            setSubmitDisabled(false);
+            return;
+        }
+
+        const response = await supabase.from("collection_centres").insert(fields);
+
+        if (response.error) {
+            setSubmitError(Errors.external);
+            setSubmitErrorMessage(response.error.message);
+            setSubmitDisabled(false);
+            return;
+        }
+
+        setSubmitError(Errors.none);
+        setSubmitErrorMessage("");
+        setSubmitDisabled(false);
+        setRefreshRequired(true);
+    };
+
+    return (
+        <CenterComponent>
+            <StyledForm>
+                {formSections.map((Card, index) => {
+                    return (
+                        <Card
+                            key={index}
+                            fields={fields}
+                            fieldSetter={fieldSetter}
+                            formErrors={formErrors}
+                            errorSetter={errorSetter}
+                        />
+                    );
+                })}
+
+                {refreshRequired ? (
+                    <RefreshPageButton />
+                ) : (
+                    <Button
+                        startIcon={<FontAwesomeIcon icon={faBuildingCircleArrowRight} />}
+                        variant="contained"
+                        onClick={submitForm}
+                        disabled={submitDisabled}
+                    >
+                        Create Collection Centre
+                    </Button>
+                )}
+
+                {/* TODO VFB-23 Improve error message formatting for request errors */}
```

Comment:
Why is this VFB-23?

Reply:
Sorry it is VFB-22

Reply:
But probably you will merge earlier than me so I will just look at your finalized message to change this